test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AssemblyAndClassInitializeTimeout.cs

```diff
@@ -0,0 +1,141 @@
+    }
+
+    [TestFixture(TestFixtureSharingStrategy.PerTestGroup)]
+    public sealed class TestAssetFixture(AcceptanceFixture acceptanceFixture) : TestAssetFixtureBase(acceptanceFixture.NuGetGlobalPackagesFolder)
+    {
+        public const string MSTestAssetName = "AssemblyAndClassInitializeTimeout";
+
+        public string MSTestTargetAssetPath => GetAssetPath(MSTestAssetName);
+
+        public override IEnumerable<(string ID, string Name, string Code)> GetAssetsToGenerate()
+        {
+            yield return (MSTestAssetName, MSTestAssetName,
+                SourceCode
+                .PatchTargetFrameworks(TargetFrameworks.All)
+                .PatchCodeWithReplace("$MicrosoftTestingPlatformVersion$", MicrosoftTestingPlatformVersion)
+                .PatchCodeWithReplace("$MicrosoftTestingPlatformExtensionsVersion$", MicrosoftTestingPlatformExtensionsVersion)
+                .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion));
+        }
+
+        private const string SourceCode = """
+#file AssemblyAndClassInitializeTimeout.csproj
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <EnableMSTestRunner>true</EnableMSTestRunner>
+    <TargetFrameworks>$TargetFrameworks$</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="MSTest.TestAdapter" Version="$MSTestVersion$" />
+    <PackageReference Include="MSTest.TestFramework" Version="$MSTestVersion$" />
+    <PackageReference Include="Microsoft.Testing.Platform" Version="$MicrosoftTestingPlatformVersion$" />
+  </ItemGroup>
+
+</Project>
+
+#file UnitTest1.cs
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+public class TestClassBase
+{
+    [ClassInitialize(inheritanceBehavior: InheritanceBehavior.BeforeEachDerivedClass)]
+    [Timeout(2000)]
+    public static async Task ClassInitBase(TestContext testContext)
+    {
+        if (Environment.GetEnvironmentVariable("TIMEOUT_BASE_CLASSINIT") == "1")
+        {
+            await Task.Delay(10_000, testContext.CancellationTokenSource.Token);
+        }
+        else
+        {
+            await Task.CompletedTask;
+        }
+    }
+
+}
+
+[TestClass]
+public class TestClass : TestClassBase
+{
+    [AssemblyInitialize]
+    [Timeout(2000)]
+    public static async Task AssemblyInit(TestContext testContext)
+    {
+        if (Environment.GetEnvironmentVariable("TIMEOUT_ASSEMBLYINIT") == "1")
+        {
+            await Task.Delay(60_000, testContext.CancellationTokenSource.Token);
+        }
+        else
+        {
+            await Task.CompletedTask;
+        }
+    }
+
+    [ClassInitialize]
+    [Timeout(2000)]
+    public static async Task ClassInit(TestContext testContext)
+    {
+        if (Environment.GetEnvironmentVariable("TIMEOUT_CLASSINIT") == "1")
+        {
+            await Task.Delay(60_000, testContext.CancellationTokenSource.Token);
```

Comment:
We probably want to fail test if cancellation doesn't happen

```suggestion
            await Task.Delay(60_000, testContext.CancellationTokenSource.Token);
            Assert.Fail("This should not be called");
```